app/controllers/users_controller.rb

```diff
@@ -0,0 +1,37 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    tokens = YnabService.new.fetch_tokens(params['code'])
+    user_data = format_user_data(tokens)
+    binding.pry
+    user = User.create!(user_data)
+    session[:user_id] = user.id
+    redirect_to signup_profile_path
+  end
```

Comment:
This create method looks much cleaner now! Might wanna take out the `binding.pry` though...